### PR TITLE
SB-65 Artifact Archive Listing

### PR DESCRIPTION
**Pull request by fuss86**

### CODE REVIEW

**Comment by fuss86** [comment]
Moved to #754

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * @author Przemyslaw Fusik
+ */
+public enum Bzip2ArchiveListingFunction
+        implements ArchiveListingFunction
+{
+
+    INSTANCE;
+
+    @Override
+    public Set<String> listFilenames(final RepositoryPath path)
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java]: I'd say it hsould be called `listFileNames` (with a capital 'N')...

Comment (fuss86) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java]: Hmm how about this https://docs.oracle.com/javase/7/docs/api/java/io/FilenameFilter.html ? Do you really want capital 'N' ?

Comment (carlspring) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/Bzip2ArchiveListingFunction.java]: Ah... I see... Okay.

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/ArchiveListingFunction.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@FunctionalInterface
+public interface ArchiveListingFunction
+{
+
+    Set<String> listFilenames(RepositoryPath path)
```
[the remark this answers is not in the document]
Comment (fuss86) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/ArchiveListingFunction.java]: It's currently named like this :P

File: strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/ArchiveListingFunction.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            throws IOException;
+
+    default Set<String> getEntriesNames(final ArchiveInputStream archiveInputStream)
```
[the remark this answers is not in the document]
Comment (fuss86) [reply strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/artifact/archive/ArchiveListingFunction.java]: Done

**Comment by fuss86** [comment]
You can merge it, @carlspring :)

**Comment by carlspring** [comment]
Thanks! :)

**Pull request merged**